### ISSUE TITLE
Now that validations are part of normal mission flow, have to wait for a long time on 'Continue'

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#1792]
When you finish a labeling mission and click 'Continue' on the 'Mission Complete' screen, you have to wait an uncomfortably long time to get the initial validation mission to load. As a user, I was like: "Did the system recognize my 'Continue' click? Did I properly click on the button? What's going on?"

At the very least, need to provide better 'Loading...' feedback to the user after clicking 'Continue' (and just generally anywhere where there can be a long load time).

**Comment by jonfroehlich** [comment]
Looks like we've talked more about this in https://github.com/ProjectSidewalk/SidewalkWebpage/issues/1799.

Including examples: https://github.com/ProjectSidewalk/SidewalkWebpage/issues/1799#issuecomment-511596221

> Here's a long load time example: https://youtu.be/8jzlYaJm8eU. The crazy thing is that I had tabbed away from this window for like ~2-3 mins then saw it was still loading and decided to open up screen recorder and then finally started recording and still had to wait for a long time! I kept video recording so you could see that the next mission load was fast... so not a consistent bug unfortunately.

**Comment by misaugstad** [comment]
So when you said "long load time" here, this means like >10 seconds? Or for a 5-10 second delay do you think we need some "loading" feedback?

**Comment by jonfroehlich** [comment]
Any perceptible delay (> 500ms) should give feedback imo.

**Comment by misaugstad** [comment]
@timvng after an audit mission, when we switch to a validation mission can you maybe switch to a "wait" cursor here?

**Comment by jonfroehlich** [comment]
Can't we just use a our "loading" gif as an overlay?

**Comment by misaugstad** [comment]
I think that in this situation, the overlaid loading gif is fine since we are already redirecting them to a new page. I think that in other instances, the loading gif as an overlay might be a bit jarring, which is why I was thinking about something more subtle like the loading icon in a status bar type idea.

**Comment by jonfroehlich** [comment]
Right... and perhaps we could do more subtle loading icon in status bar near or inline with the "Project Sidewalk" version line on bottom?

**Comment by misaugstad** [comment]
Yeah I like that. And that location will work for the validation page as well. And almost all instances of this would be on audit/validation. Maybe we would consider something in the navbar though, maybe to the right of the sign in button? The only rationale for that is that the navbar is consistent across all pages of the website.

**Comment by misaugstad** [comment]
>there is about a ~3-5 sec delay after completing a validation mission before being able to click continue, I don't think it's too excessive.

This seems fine to me. We do have another ticket that's focused on a specific speedup that we can do (#3004). Other than that, there aren't easy speedups available to us. And the 3-5 seconds feels fine to me too.

>Any perceptible delay (> 500ms) should give feedback imo.

This comment is why I hadn't closed this issue before. But at the very least, we gray out the continue button until the new labels to validate have loaded, and I've set up the "wait" cursor to show on the button while it's grayed out. This should be enough feedback for now.

**Comment by misaugstad** [comment]
It might feel a little jarring! I think what we have is fine for now, but I'm sure there's a way we could do it later
